src/main/java/org/folio/circulation/services/EventPublisher.java

```diff
@@ -129,14 +129,24 @@ public CompletableFuture<Result<CheckInContext>> publishItemCheckedInEvents(
     runAsync(() -> userRepository.getUser(checkInContext.getLoggedInUserId())
       .thenCombineAsync(loanRepository.findLastLoanForItem(checkInContext.getItem().getItemId()), (userResult, lastLoan) -> {
         if (nonNull(lastLoan.value())) {
-          return userRepository.getUser(lastLoan.value().getUserId())
-            .thenApply(userFromLastLoan -> Result.succeeded(pubSubPublishingService.publishEvent(LOG_RECORD.name(),
-              mapToCheckInLogEventContent(checkInContext, userResult.value(),
-                checkInContext.isInHouseUse() ? null : userFromLastLoan.value()))));
+          return loanRepository.findOpenLoanForItem(checkInContext.getItem()).thenApply(openLoanResult -> {
+            if(nonNull(openLoanResult.value())) {
+              userRepository.getUser(lastLoan.value().getUserId())
+                .thenApply(userFromLastLoan -> Result.succeeded(pubSubPublishingService.publishEvent(LOG_RECORD.name(),
+                  mapToCheckInLogEventContent(checkInContext, userResult.value(),
+                    checkInContext.isInHouseUse() ||
+                      checkInContext.getRequestQueue().getRequests().isEmpty() ? null :
+                      userFromLastLoan.value()
+                  ))));
+            }
+            return Result.succeeded(pubSubPublishingService.publishEvent(LOG_RECORD.name(),
```

Comment:
This will return a `Result<CompletableFuture<...>>` which then will be wrapped into another `CompletableFuture` in line 132